### DISCUSSION TITLE
Crazy Question

### DISCUSSION THREAD

**Discussion by elosoman** [Ultrawipf/OpenFFBoard#32] [General]
I'm been looking over your project for a while now. Very nice!

I've tried JLCPCB but they are out of stock on parts. I'm working with PcbWay get 5 of your boards made.

Full disclosure ... i'm that crazy boat guy from discord who is trying to make a boat throttle control. So as I continue to understand and setup my system, could I ask your thoughts about implementing the following?

1) Does the CAN implementation transmit messages based on changes to the inputs? Can it receive messages to change its settings? Or is all CAN messaging a roll your own. My thoughts are to use can to network 3 of these boards via can to function as a unit.

2) How could you detect pressure being applied to an end stop? The thinking here is to create a neutral position on the wheel. Kind of 3 modes of operation 
 a. neutral - 5 degrees freedom of rotation. Force against the end stop would switch mode to either forward or reverse.
 b. forward - 0 to 90 degrees of freedom. When return to 0 switch to neutral.
 c. reverse - 1 to -90 degrees of neutral. When return to 0 switch to neutral.
 
Good work and thanks for you time!
Ben

**Comment by Ultrawipf** [comment]
At the moment the CAN bus is only used to control certain motor drivers but nothing stops you from implementing a custom main class for your application if its very special and requires that.

I think if you are not using HID FFB effects then you would need a custom mainclass and make your own effects. Then you could also send out positions and inputs with a custom CAN protocol and use the already implemented motor driver, encoder, CAN and usb features to simplify the  communication.
It might be a bit more work but the firmware is designed that it should be as simple as possible without having to develop a fully custom firmware.

**Comment by laverdone** [comment]
Hi a all, congratulation for Your Project! I have a questione: this board/firmware are supporter under Linux with FFB? Thanks
